### DISCUSSION TITLE
Validation Callbacks

### DISCUSSION THREAD

**Discussion by dlim87** [excid3/noticed#247] [Q&A]
This is probably something really simple rails-y convention that I'm missing, but how would you go about using the `before_deliver` callback to do validations (beyond the existing 'does the param exist' validations), and abort the notification if the conditions aren't met? 

Right now I'm doing a bit of a hack, applying the `if: <method>` check on each `deliver_by`, but that doesn't seem like the right way to do it

**Comment by excid3** [comment]
I would use the `if:` option. Or if you know ahead of time, don't send the notification at all?

**Comment by benbartling** [comment]
I believe what @dlim87 is asking for is a way to abort the notification in the callback. I believe doing a `throw(:abort)` should work.

```ruby
before_deliver -> { throw(:abort) }
```
